backend/internal/observability/service.go

```diff
@@ -0,0 +1,270 @@
+	"github.com/asgardeo/thunder/internal/observability/subscriber/defaultsubscriber"
+	"github.com/asgardeo/thunder/internal/system/config"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+const loggerComponentName = "ObservabilityService"
+
+// Service provides observability event publishing functionality.
+// This is the main entry point for the observability system.
+// It manages the lifecycle of the publisher, which in turn manages subscribers.
+//
+// Architecture:
+//   - Service (High-level): Manages lifecycle, configuration, initialization
+//   - CategoryPublisher (Low-level): Implements event publishing logic
+//   - Subscribers (Low-level): Consume events
+//
+// The service layer owns the singleton pattern and initialization logic,
+// while the publisher layer provides the implementation.
+type Service struct {
+	publisher publisher.CategoryPublisher
+	logger    *log.Logger
+	config    *Config
+	enabled   bool
+}
+
+var (
+	serviceInstance *Service
+	serviceOnce     sync.Once
+)
+
+// GetService returns the singleton instance of the observability service.
+// Uses default configuration. For custom configuration, use InitializeWithConfig first.
+func GetService() *Service {
+	serviceOnce.Do(func() {
+		serviceInstance = NewService(nil)
+	})
+	return serviceInstance
+}
+
+// InitializeWithConfig initializes the observability service with custom configuration.
+// Must be called before GetService to take effect.
+// Returns error only if FailureMode is "strict", otherwise logs error and disables observability.
+func InitializeWithConfig(cfg *Config) (*Service, error) {
+	var svc *Service
+
+	serviceOnce.Do(func() {
+		svc = newServiceWithConfig(cfg)
+		serviceInstance = svc
+	})
+
+	if serviceInstance == nil {
+		return nil, errors.New("failed to initialize observability service")
+	}
+
+	return serviceInstance, nil
+}
+
+// NewService creates and initializes a new observability service with the given configuration.
+func NewService(cfg *Config) *Service {
+	svc := newServiceWithConfig(cfg)
+	return svc
+}
+
+// newServiceWithConfig creates and initializes a new observability service.
+func newServiceWithConfig(cfg *Config) *Service {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName))
+
+	// Use default config if none provided
+	if cfg == nil {
+		cfg = DefaultConfig()
+	}
+
+	// Check if observability is disabled
+	if !cfg.Enabled {
+		logger.Info("Observability service is disabled by configuration")
+		return &Service{
+			publisher: nil,
+			logger:    logger,
+			config:    cfg,
+			enabled:   false,
+		}
+	}
+
+	logger.Info("Initializing observability service",
+		log.String("outputType", cfg.Output.Type))
+
+	// Create event bus (no queue needed)
+	pub := publisher.NewCategoryPublisher()
+
+	svc := &Service{
+		publisher: pub,
+		logger:    logger,
+		config:    cfg,
+		enabled:   true,
+	}
+
+	// Initialize default subscribers
+	svc.initializeDefaultSubscribers()
```

Comment:
Shouldn't this be initializeSubscribers ?
```suggestion
	svc.initializeSubscribers()
```